src/Cli/dotnet/ParseResultExtensions.cs

```diff
@@ -135,25 +137,12 @@ public static string[] GetSubArguments(this string[] args)
             return subargs.Concat(runArgs).ToArray();
         }
 
-        private static string GetSymbolResultValue(ParseResult parseResult, SymbolResult symbolResult)
+        private static string GetSymbolResultValue(ParseResult parseResult, SymbolResult symbolResult) => symbolResult switch
```

Comment:
This is a very clean pattern 😄

I want to make sure I understand this change correctly. What is `GetSymbolResultValue` used for? My understanding is that this will get the first token in the args if it is passed an `argResult`. Where as before it looks like it just.. got the value of the argument token. Was SCL changed such that it gets the entire set of args when before it just got the argument token?

Reply:
> What is `GetSymbolResultValue` used for?

This method went through various versions as Chet and I both investigated it. It is only used in `RootSubCommandResult`. In that context, it converts the root command's children (passed in here as `symbolResult`) to a valid string value. Hence the name `GetSymbolResultValue` of taking a SymbolResult and pulling the value (as a string) out of it.

> Was SCL changed such that it gets the entire set of args when before it just got the argument token?

In this PR, we deleted the `.Token()` extension method which was the only method in the file `CommandLineValidationMessages.cs`. So, I just deleted the whole file. The issue with that method was the `argResult.GetValueOrDefault<string>()`. This is the method that changed in the new S.CL. Previously, it would allow the value to be `.ToString()` of the value from the token. Now, it will **ONLY** return a `<string>` if the value is actually a string. An example that causes a runtime error is `--verison 1.0.0`, because the value, `1.0.0`, is not a string value. It is a `System.Version`. It used to automatically turn that into a `"1.0.0"`. Now, it doesn't.

Chet said that `.Token()` doesn't need to exist so this method and the `GetOptionTokenOrDefault` local function I added encompass what is actually required. Here, the logic I wrote for `GetSymbolResultValue` is nearly identical to the if-elses that were originally there, except it doesn't compare against `Token().Type` but compares directly against the `symbolResult` type.

- The `== default` check only happened if the type wasn't `CommandResult`, `OptionResult`, or `ArgumentResult`. There is only one other type, `DirectoryResult`, I believe that would actually cause this condition to happen. I made this the new default (`_` in switch expression) logic for this method.
- The `CliTokenType.Command` is just checking if it is a `CommandResult`, so we do that same logic and return `Command.Name`.
- The `CliTokenType.Argument` is just checking if it is a `ArgumentResult` so we do that same logic and return the result of the `DotnetSubCommand` as a string value.
- After looking at this logic, I don't believe there is any way for the *else* of `string.Empty` could have ever been returned. So, I've removed it from the switch expression.

Reply:
Talked about this offline and it makes sense to me now, thanks :)